src/adapter/src/coord/appends.rs

```diff
@@ -278,99 +286,114 @@ impl<S: Append + 'static> Coordinator<S> {
                 (id, updates, advance_to)
             })
             .collect();
-        if !appends.is_empty() {
-            self.controller
-                .storage_mut()
-                .append(appends)
-                .expect("invalid updates")
-                .await
-                .expect("One-shot shouldn't fail")
-                .unwrap();
+
+        let append_fut = self
+            .controller
+            .storage_mut()
+            .append(appends)
+            .expect("invalid updates");
+        if should_block {
+            append_fut.await.expect("One-shot shouldn't fail").unwrap();
+            self.group_commit_apply(timestamp, responses, true, write_lock_guard)
+                .await;
+        } else {
+            let internal_cmd_tx = self.internal_cmd_tx.clone();
+            task::spawn(|| "group_commit_apply", async move {
+                append_fut.await.expect("One-shot shouldn't fail").unwrap();
+                if let Err(e) = internal_cmd_tx.send(Message::GroupCommitApply(
+                    timestamp,
+                    responses,
+                    write_lock_guard,
+                )) {
+                    warn!("Server closed with non-responded writes, {e}");
+                }
+            });
+        }
+    }
+
+    /// Applies the results of a completed group commit. The global timestamp will be advanced to
```

Comment:
"The global timestamp" -> "read timestamp"? I'm not actually sure what the global timestamp is tbh. This probably means something like "the read timestamp of the timeline(s) of the writes"?

Reply:
I think there's still a lot of language in the comments in the Coord that assumes there's only a single timeline. The global timestamp generally refers to the timestamp of the real time timeline.

How does this sound
> The read timestamp of the timeline containing user tables will be advanced to the timestamp of the completed write, the read hold on the timeline containing user tables is advanced to the new time, and responses are sent to all waiting clients.